### PR TITLE
Add entity ID being browsed to URL of media browser

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Store the selected media player in the URL. This is step 1 in addressing #10950

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/media-browser/ha-panel-media-browser.ts [hunk 1]
```diff
@@ -74,11 +84,29 @@ class PanelMediaBrowser extends LitElement {
...
+      this._entityId = routePlayer;
+    }
+  }
+
   private _showSelectMediaPlayerDialog(): void {
     showSelectMediaPlayerDialog(this, {
       mediaSources: this._mediaPlayerEntities,
       sourceSelectedCallback: (entityId) => {
-        this._entityId = entityId;
+        navigate(`/media-browser/${entityId}`, { replace: true });
```
Comment (zsarnett) [inline src/panels/media-browser/ha-panel-media-browser.ts:109]: Do we want to replace here? When clicking back is the result as expected?

File: src/panels/media-browser/ha-panel-media-browser.ts [hunk 2]
```diff
@@ -74,11 +84,29 @@ class PanelMediaBrowser extends LitElement {
...
   }
 
+  public updated(changedProps: PropertyValues): void {
+    super.updated(changedProps);
+
+    if (!changedProps.has("route")) {
+      return;
+    }
+
+    if (this.route.path === "") {
```
Comment (zsarnett) [inline src/panels/media-browser/ha-panel-media-browser.ts:94]:
```suggestion
    if (this.route.path === "" || this.route.path === "/") {
```

Comment (zsarnett) [reply src/panels/media-browser/ha-panel-media-browser.ts:94]: This Safe guards removing everything after the slash but leaving the slash

Should we also check to see if the entityId is valid?

image

Comment (balloob) [reply src/panels/media-browser/ha-panel-media-browser.ts:94]: I think that error being shown is just fine. It means people have messed with the URL.

Comment (zsarnett) [reply src/panels/media-browser/ha-panel-media-browser.ts:94]: But the error isn't very user-friendly. Example: If you tried to navigate to a page that doesn't exist for some websites then you are redirected to the home page. (there are 404 examples) but its probably better to push them to the entityId we have from the last time they chose a correct one if we can

**Review by zsarnett — APPROVED** [review]

File: src/panels/media-browser/ha-panel-media-browser.ts (hunk 2 above)
Comment (balloob) [reply src/panels/media-browser/ha-panel-media-browser.ts:94]: Right, but there is no way we navigate users to pages that don't exist today.

**Pull request merged**